<?xml version="1.0"?>
<launch>
    <arg name="dataset" value="kinetic_realtime_"/>
    <arg name="visualization" value="false"/>
    <arg name="publish_refined_keyframe" value="true"/>
    <arg name="publish_increnmental_keypose_with_covariance" value="true"/>
    <arg name="publish_twist_with_covariance" value="false"/>
    <arg name="parent_frame" value="local_origin"/>
    <arg name="child_frame" value="camera_link"/>
    <arg name="queue_size_cloud" value="1"/>
    <arg name="queue_size_imu" value="10"/>
    <arg name="queue_size_sync" value="50"/>
    <param name="/map/resolution" value="0.005" />
    <param name="/axonometric/image_height" value="360" />
    <param name="/axonometric/image_width" value="480" />
    <param name="use_sim_time" value="false" />

    <!-- rotation must be [pitch, roll, yaw] -->
    <!-- this is facing front camera -->
    <rosparam param="rotation/imu"> [1.5707963, 0, -1.5707963] </rosparam>

    <!-- this is for facing down camera -->
    <!-- <rosparam param="rotation/imu"> [0, 3.1415926, 1.5707963] </rosparam> -->

    <!-- this is for facing right camera -->
    <!-- <rosparam param="rotation/imu"> [0, 1.5707963, 3.1415926] </rosparam> -->

    <!-- <include file="$(find viconxbee)/launch/viconxbee.launch"/> -->

    <include file="$(find ni_slam)/launch/kinect_imu.launch" />

    <node pkg="ni_slam" type="ni_slam_node" name="ni_slam_node" output="screen">
        <remap from="/camera/rgb/points" to="/camera/depth_registered/points"/>
        <!-- <remap from="/imu/data" to="/imu/vicon/data"/> -->
        <param name="saveposes/filename_prefix" value="$(find ni_slam)/bag/$(arg dataset)" />
        <!--comment above line if you don't want to save any files, 
            or any existing files would be replaced.-->
        
        <param name="adaptive/mu" value="0.8" />
        <param name="topic/time_diff" value="0.02" />
        <param name="train/psr_bound" value="70.0" />
       <param name="frame_name/parent" value="$(arg parent_frame)" />
        <param name="frame_name/child" value="$(arg child_frame)" />
        <param name="debug/visualization" value="$(arg visualization)" />
        <param name="refined_frame/publish_flag" value="$(arg publish_refined_keyframe)" />
        <param name="incremental_keypose_with_covariance/publish_flag" value="$(arg publish_increnmental_keypose_with_covariance)" />
        <param name="twist_with_covariance/publish_flag" value="$(arg publish_twist_with_covariance)" />
        <param name="queue_size/cloud" value="$(arg queue_size_cloud)" />
        <param name="queue_size/imu" value="$(arg queue_size_imu)" />
        <param name="queue_size/sync" value="$(arg queue_size_sync)" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_ni_slam" output="screen" ns="est" >
        <param name="/target_frame_name" value="$(arg parent_frame)" />
        <param name="/source_frame_name" value="$(arg child_frame)" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>

<!--     <node pkg="rosbag" type="record" name = "map_record_node" output="screen" args="-o $(find ni_slam)/bag/map/realsense /ni_slam_node/refined/image_color /ni_slam_node/refined/image_depth /ni_slam_node/posestamped_key">
    </node> -->

<!--     <node pkg="rosbag" type="record" name = "realsense_imu_bag" output="screen" args="-o $(find ni_slam)/bag/iot/realsense /camera/depth/image /camera/depth/camera_info /camera/rgb/camera_info /camera/rgb/image_color /imu/data /viconxbee_node/mocap/pose /viconxbee_node/mocap/imu /tf">
    </node> -->

</launch>
